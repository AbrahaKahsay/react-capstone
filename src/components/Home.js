import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { coinList, coinFilter } from '../redux/home/Coin';
import Card from './Card';
import './styles/Home.css';

const Home = () => {
  const dispatch = useDispatch();
  const cList = useSelector((store) => store.coinReducer);
  useEffect(() => {
    dispatch(coinList());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const changeHandler = (e) => {
    dispatch(coinFilter(e.target.value));
  };

  const searchHandler = (e) => {
    e.preventDefault();
    window.location.href = `/detail/${e.target.coinSearch.value}`;
  };

  return (
    <section>
      <div className="info-list">
        <div className="options">
          <select className="filter" onChange={changeHandler}>
            <option value="price"> Rank </option>
            <option value="name"> Name </option>
            <option value="vol"> Price </option>
          </select>
        </div>
        <div className="forms">
          <form onSubmit={searchHandler}>
            <input type="text" name="coinSearch" placeholder="Fill Crypto name ..." required />
            <input type="submit" value="Search" />
          </form>
        </div>
      </div>
      <ul className="coinList">
        {cList.map((coin) => (
          <Card
            key={coin.id}
            id={coin.id}
            symbol={coin.symbol}
            price={coin.priceUsd.substring(0, 8)}
            change={coin.changePercent24Hr}
          />
        ))}
      </ul>
    </section>
  );
};

export default Home;
