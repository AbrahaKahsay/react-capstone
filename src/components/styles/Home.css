.coinList {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    text-align: center;
    margin: 0;
    padding: 0;
}

.card:nth-child(4n),
.card:nth-child(4n+1) {
    background-color: rgb(54,88,148);
}

.card {
    background-color: rgb(67,105,178);
    color: white;
    margin: 0;
}
.info-list {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: rgb(88,137,228);
    height: 50px;
}

.forms {
    line-height: 20px;
}

.options {
    line-height: 20px;
}