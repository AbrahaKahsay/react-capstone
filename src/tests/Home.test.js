import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Jest Snapshot testing suite', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
