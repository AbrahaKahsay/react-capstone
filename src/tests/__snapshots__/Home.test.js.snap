// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<section>
  <div
    class="info-list"
  >
    <div
      class="options"
    >
      <select
        class="filter"
      >
        <option
          value="price"
        >
           Rank 
        </option>
        <option
          value="name"
        >
           Name 
        </option>
        <option
          value="vol"
        >
           Price 
        </option>
      </select>
    </div>
    <div
      class="forms"
    >
      <form
        class="search-form"
      >
        <input
          name="coinSearch"
          placeholder="Fill Crypto name ..."
          required=""
          type="text"
        />
        <input
          type="submit"
          value="Search"
        />
      </form>
    </div>
  </div>
  <ul
    class="coinList"
  />
</section>
`;
